app/views/videos/show.html.haml

```diff
@@ -10,8 +10,7 @@
         %p= @video.description
         .actions
           %a.btn.btn-primary(href="") Watch Now
-          / %a.btn.btn-default(href="") + My Queue
-          =link_to "+ My Queue", queue_items_path(queue_item: { user_id: @current_user, video_id: @video }), method: :post, class: "btn btn-default"
+          =link_to "+ My Queue", queue_items_path(queue_item: { user_id: @current_user, video_id: @video }), method: :post, class: "btn btn-default" unless @current_user.in_queue?(@video)
```

Comment:
This part is confusing: `queue_items_path(queue_item: { user_id: @current_user, video_id: @video })`.

Confusion points are on: you are passing an object and annotating it as an `something_id`. stick with id to id, object to object. Next confusion is that you actually pass @current_user back to controller. You don't need that, your controller should know who is the current_user, this way you give the chance to User to mess up with your form, because he can influence params in HTML code.